include/actionspace/ObjectOrientedActionSpace.hpp

```diff
@@ -98,21 +98,30 @@ class ObjectOrientedActionSpace : public virtual ActionSpace {
         /// \param heading_offset : The angular distance from front of
         ///     the object, indicating which side of the object the action
         ///     is being applied to (radians)
-        explicit GenericAction(
+        explicit Action(
             const double& speed,
             const double& edge_offset,
             const double& aspect_ratio,
-            const double& heading_offset) : \
-            m_speed(speed),
-            m_edge_offset(edge_offset),
-            m_aspect_ratio(aspect_ratio),
-            m_heading_offset(heading_offset) {}
+            const double& heading_offset);
 
         double speed() const { return m_speed; }
         double aspect_ratio() const { return m_aspect_ratio; }
         double edge_offset() const { return m_edge_offset; }
         double heading_offset() const { return m_heading_offset; }
 
+        /// Documentation inherited
+        /// In this class the action vector is in format
```

Comment:
The action vector is in the following format: [speed, aspect_ratio, edge_offset, heading_offset]